src/components/sbb-journey-header/sbb-journey-header.tsx

```diff
@@ -10,75 +10,60 @@ import { InterfaceJourneyHeaderAttributes } from './sbb-journey-header.custom';
   @Prop() public destination!: string;
 
-  /**
-   * Is the journey a round trip. If so it gets indicated through a roundtrip
-   * icon
-   */
-  @Prop() public isRoundTrip?: boolean;
+  /** Whether the journey is a round trip. If so, the icon changes to a round-trip one. */
+  @Prop() public roundTrip?: boolean;
 
   /**
-   * Journey header markup:
-   * Depends on the context where the journey will be used but it is important
-   * to pick the correct markup element to match to correct semantics
+   * Journey header markup: depending on the context where it will be used,
+   * it is important to pick the correct markup element to match the correct semantics.
    */
-  @Prop() public markup?: InterfaceJourneyHeaderAttributes['markup'] = 'span';
-
-  /** Journey header appearance */
-  @Prop() public appearance?: InterfaceJourneyHeaderAttributes['appearance'] = 'primary';
-
-  /** Journey header size */
-  @Prop() public size?: InterfaceJourneyHeaderAttributes['size'] = '5';
-
-  public render(): JSX.Element {
-    const journeyIcon = this.isRoundTrip ? iconRoundTrip : iconOneWay;
+  @Prop() public level?: InterfaceJourneyHeaderAttributes['level'] = '3';
 
-    /*
-     * Connection text
-     * @Todo i18n handling
-     * @Todo Check if anybody did think about `via` connections yet? Or will
-     * this component never show any `via` connections?
-     */
-    const connectionTextOrigin = 'Connection from';
-    const connectionTextDestination = 'to';
-    const connectionTextRoundtrip = this.isRoundTrip ? `and back to ${this.origin}` : '';
+  /** Negative coloring variant flag. */
+  @Prop({ reflect: true }) public negative = false;
 
-    const TAGNAME = `${this.markup}`; // eslint-disable-line @typescript-eslint/no-unused-vars
+  /** Journey header size. */
+  @Prop({ reflect: true }) public size?: InterfaceJourneyHeaderAttributes['size'] = 'm';
 
-    const className = `journey-header journey-header--${this.appearance} journey-header--size-${this.size}`;
+  @State() private _currentLanguage = documentLanguage();
 
-    const currentWritingMode = getDocumentWritingMode();
+  @Listen('sbbLanguageChange', { target: 'document' })
+  public handleLanguageChange(event: SbbLanguageChangeEvent): void {
+    this._currentLanguage = event.detail;
+  }
 
-    const attrs = {
-      class: className,
-    };
+  public render(): JSX.Element {
+    const iconName = this.roundTrip ? 'arrows-left-right-small' : 'arrow-long-right-small';
 
     return (
-      <TAGNAME {...attrs} dir={currentWritingMode}>
-        <span class="connection-text-origin connection--visually-hidden">
-          {connectionTextOrigin}
-        </span>
-        <span class="origin">{this.origin}</span>
-        <span class="icon" innerHTML={journeyIcon}>
-          <span class="connection-text-destination connection--visually-hidden">
-            {connectionTextDestination}
+      <sbb-title level={this.level} negative={this.negative}>
```

Comment:
You need to transform the size property into the visualLevel property of the sbb-title. I don't know the actual sizes but something like 
```suggestion
      <sbb-title level={this.level} negative={this.negative} visual-level={this.size === 'm' ? '5' : '3'}>
```